src/main/java/core/basesyntax/strategy/handlers/PurchaseHandler.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.strategy.handlers;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.StrategyCalculator;
+
+public class PurchaseHandler implements StrategyCalculator {
+    @Override
+    public void calculate(FruitTransaction fruitTransaction) {
+        Storage.STORAGE.put(fruitTransaction.getFruit(),
+                Storage.STORAGE.get(fruitTransaction.getFruit())
+                - fruitTransaction.getQuantity());
```

Comment:
What will happen is fruitTransaction.getQuantity() is bigger then Balace?
And also you can use merge method:
Storage.STORAGE.merge(fruitTransaction.getFruit(), -fruitTransaction.getQuantity(), Integer::sum);
you can do the same for Return and Supply.